src/pipecat/processors/aggregators/openai_llm_context.py

```diff
@@ -102,6 +103,34 @@ def set_tools(self, tools: List[ChatCompletionToolParam] | NotGiven = NOT_GIVEN)
             tools = NOT_GIVEN
 
         self.tools = tools
+    
+    async def call_function(
+            self,
+            f: callable,
+            *,
+            function_name: str,
+            tool_call_id: str,
+            arguments: str,
+            llm: FrameProcessor) -> None:
+
+        # Push a SystemFrame downstream. This frame will let our assistant context aggregator
+        # know that we are in the middle of a function call. Some contexts/aggregators may
+        # not need this. But some definitely do (Anthropic, for example).
+        await llm.push_frame(FunctionCallInProgressFrame(
+            function_name=function_name,
+            tool_call_id=tool_call_id,
+            arguments=arguments,
+        ))
+
+        # Define a callback function that pushes a FunctionCallResultFrame downstream.
+        async def function_call_result_callback(result):
+            await llm.push_frame(FunctionCallResultFrame(
+                function_name=function_name,
+                tool_call_id=tool_call_id,
+                arguments=arguments,
+                result=result))
+        await f(function_name=function_name, tool_call_id=tool_call_id, arguments=arguments,
+                context=self, result_callback=function_call_result_callback)
```

Comment:
why not?

```
result = await f(function_name=function_name, tool_call_id=tool_call_id, arguments=arguments, context=self)

await llm.push_frame(FunctionCallResultFrame(
                function_name=function_name,
                tool_call_id=tool_call_id,
                arguments=arguments,
                result=result))
```

Reply:
There's no way for the user to pass a result callback as far as I could see.

Reply:
So maybe this is me not understanding Python async well enough ...

But I think awaiting here for a result blocks all frame processing by the llm until the function call returns. I was trying to avoid that.